ui/components/app/collectible-details/collectible-details.js

```diff
@@ -0,0 +1,229 @@
+    ({ address: contractAddress }) =>
+      isEqualCaseInsensitive(contractAddress, address),
+  )?.name;
+  const selectedAccountName = useSelector(
+    (state) => getSelectedIdentity(state).name,
+  );
+  const collectibleImageURL = getAssetImageURL(image, ipfsGateway);
+  const dispatch = useDispatch();
+
+  const onRemove = () => {
+    dispatch(removeAndIgnoreCollectible(address, tokenId));
+    history.push(DEFAULT_ROUTE);
+  };
+
+  const getOpenSeaLink = () => {
+    switch (currentNetwork) {
+      case MAINNET_CHAIN_ID:
+        return `https://opensea.io/assets/${address}/${tokenId}`;
+      case POLYGON_CHAIN_ID:
+        return `https://opensea.io/assets/matic/${address}/${tokenId}`;
+      case GOERLI_CHAIN_ID:
+      case KOVAN_CHAIN_ID:
+      case ROPSTEN_CHAIN_ID:
+      case RINKEBY_CHAIN_ID:
+        return `https://testnets.opensea.io/assets/${address}/${tokenId}`;
+      default:
+        return null;
+    }
+  };
+
+  const openSeaLink = getOpenSeaLink();
+  return (
+    <>
+      <AssetNavigation
+        accountName={selectedAccountName}
+        assetName={collectibleContractName}
+        onBack={() => history.push(DEFAULT_ROUTE)}
+        optionsButton={
+          <CollectibleOptions
+            onViewOnOpensea={
+              openSeaLink
+                ? () => global.platform.openTab({ url: openSeaLink })
+                : null
+            }
+            onRemove={onRemove}
+          />
+        }
+      />
+      <Box padding={[4, 2, 4, 2]}>
+        <div className="collectible-details">
+          <Box margin={3} padding={2} justifyContent={JUSTIFY_CONTENT.CENTER}>
+            <img
+              className="collectible-details__image"
+              src={collectibleImageURL}
+            />
+          </Box>
+          <Box
+            margin={3}
+            flexDirection={FLEX_DIRECTION.COLUMN}
+            width={
+              getEnvironmentType() === ENVIRONMENT_TYPE_POPUP
+                ? BLOCK_SIZES.THREE_FOURTHS
+                : BLOCK_SIZES.HALF
+            }
+          >
+            <Typography
+              color={COLORS.BLACK}
+              variant={TYPOGRAPHY.H4}
+              fontWeight={FONT_WEIGHT.BOLD}
+            >
+              {name}
+            </Typography>
+            <Typography
+              color={COLORS.UI3}
+              variant={TYPOGRAPHY.H5}
+              boxProps={{ marginTop: 2, marginBottom: 3 }}
+              overflowWrap={OVERFLOW_WRAP.BREAK_WORD}
+            >
+              {`#${tokenId}`}
+            </Typography>
+            <Typography
+              color={COLORS.BLACK}
+              variant={TYPOGRAPHY.H6}
+              fontWeight={FONT_WEIGHT.BOLD}
+              className="collectible-details__description"
+            >
+              {t('description')}
+            </Typography>
+            <Typography color={COLORS.UI3} variant={TYPOGRAPHY.H6}>
+              {description}
```

Comment:
Since OpenSea supports Markdown for descriptions, is it worth running this through a simple Markdown parser only for OpenSea NFTs? Not sure if that was discussed elsewhere or not. Also fine if this is more of a V2 thing.

Reply:
hmmm possibly. We ought to do some research here. In any case lets include whatever's found in v2 though since it will require some effort in the controllers to identify and store NFT's according to where we got their data (directly from the chain vs OpenSea).